analyze/analyze.rs

```diff
@@ -289,43 +291,55 @@ impl traits::Emit for Paths {
             seen: &mut BTreeSet<ir::Id>,
             table: &mut Table,
             depth: usize,
+            max_depth: usize,
+            max_row: usize,
             id: ir::Id,
         ) {
-            if seen.contains(&id) || items.meta_root() == id {
-                return;
-            }
+            if max_depth != depth && max_row != 0 {
+                if seen.contains(&id) || items.meta_root() == id {
```

Comment:
Again, rather than indenting everything, lets fold the new checks into this existing early return.